### PR TITLE
Payments: use (and show) deposit balance

### PR DESCRIPTION
**Pull request by chfast**
The deposit balance (ethers got from computations) is used as the first choise for outgoing payments.

### CODE REVIEW

File: gnr/gnrapplicationlogic.py [hunk 1]
```diff
@@ -198,12 +198,11 @@ def update_payments_view(self):
         self.client.get_balance().addCallback(self._update_payments_view)
 
     def _update_payments_view(self, result_tuple):
-        b, ab = result_tuple
+        b, ab, deposit = result_tuple
```
Comment (badb) [inline gnr/gnrapplicationlogic.py:201]: Throws exception when transaction system is turn off

File: golem/ethereum/paymentprocessor.py [hunk 2]
```diff
@@ -109,6 +110,21 @@ def balance(self, refresh=False):
...
+            addr = keys.privtoaddr(self.__privkey)
+            r = self.__client.call(_from=addr.encode('hex'),
+                                   to=self.BANK_ADDR.encode('hex'),
+                                   data=data.encode('hex'),
+                                   block='pending')
+            if r is None or r == '0x':
+                self.__deposit = 0
+            else:
+                self.__deposit = int(r, 16)
+            log.info("Deposit: {}".format(self.__deposit / float(10**18)))
```
Comment (badb) [inline golem/ethereum/paymentprocessor.py:125]: Can you use  ETH from golem.core.common here?

File: gnr/gnrapplicationlogic.py (hunk 1 above)
Comment (chfast) [inline gnr/gnrapplicationlogic.py:201]: Fixed.

File: golem/ethereum/paymentprocessor.py (hunk 2 above)
Comment (chfast) [inline golem/ethereum/paymentprocessor.py:125]: Not the best idea. We need a function or a type to represent ETH amount. First try should be pyethereum library.

Comment (badb) [inline golem/ethereum/paymentprocessor.py:125]: https://github.com/ethereum/pyethereum/blob/develop/ethereum/utils.py#L408 denoms?

Comment (chfast) [inline golem/ethereum/paymentprocessor.py:125]: Yep. I'm working on it. But will submit as a separated PR.

**Pull request merged**